packages/workflow/src/logs.ts

```diff
@@ -72,12 +73,16 @@ export const log: WorkflowLogger = Object.fromEntries(
       (message: string, attrs?: Record<string, unknown>) => {
         const activator = assertInWorkflowContext('Workflow.log(...) may only be used from workflow context.');
         const getLogAttributes = composeInterceptors(activator.interceptors.outbound, 'getLogAttributes', (a) => a);
+        const updateInfo = currentUpdateInfo();
         return loggerSink[level](message, {
           // Inject the call time in nanosecond resolution as expected by the worker logger.
           [LogTimestamp]: activator.getTimeOfDay(),
           sdkComponent: SdkComponent.workflow,
           ...getLogAttributes(workflowLogAttributes(activator.info)),
           ...attrs,
+          // Add update info if it exists
+          ...(updateInfo && { updateId: updateInfo.id }),
```

Comment:
Let's use a single object spread operation for both id and name.

Something like that:
```suggestion
          ...(updateInfo && { updateId: updateInfo.id, updateName: updateInfo.name }),
```

Reply:
Thinking more about this, I'd say we should do this in `workflowLogAttributes()`, for coherency. That behavior will be visible to user interceptors.

Only problem though is that `workflowLogAttributes()` may be called from outside the workflow sandbox, so you will need to add a call to `inWorkflowContext()` first to not attempt this in other cases.

That could give something like this:

```
export function workflowLogAttributes(info: WorkflowInfo): Record<string, unknown> {
  const attributes = {
    namespace: info.namespace,
    taskQueue: info.taskQueue,
    workflowId: info.workflowId,
    runId: info.runId,
    workflowType: info.workflowType,
  };

  // Add details about the current update if appropriate
  if (inWorkflowContext()) {
    const updateInfo = currentUpdateInfo();
    if (updateInfo) {
      attributes['updateId'] = updateInfo.id;
      attributes['updateName'] = updateInfo.name;
    }
  }

  return attributes;
}
```

Reply:
Sounds good, updated with the snippet above.

> Only problem though is that workflowLogAttributes() may be called from outside the workflow sandbox, so you will need to add a call to inWorkflowContext() first to not attempt this in other cases.

From the code references, does this only occur when a worker creates a new workflow?
https://github.com/temporalio/sdk-typescript/blob/sdk-t-add-update-info-to-logging-context/packages/worker/src/worker.ts#L1313

Reply:
> From the code references, does this only occur when a worker creates a new workflow?

Probably, yeah. This is used to provide some basic attributes very early during workflow initialization and in case starting the workflow fail for whatever reason.

Arguably, we could duplicate that code, once in Worker knowing that we are not in workflow context, and once in workflows/logs.ts where we'd know we are in workflow context, but I think that in this case it is best to keep this in a single place, to avoid eventual inconsistencies whenever we add extra workflow log attributes.